### DISCUSSION TITLE
Option group onkey[up/down/press]? How to intercept them?

### DISCUSSION THREAD

**Discussion by labs20** [quasarframework/quasar#12006] [General - Components / Directives / etc] [answered]
Hi.

I need to intercept the ENTER key when pressed while a option group has focus but I'm not being able to as it seems that the comp stops the key events to propagate.

If I'm allowed to give a constructive criticism, I'm a big fan of quasar since pre version 1, and have been myself a contributor from time to time and convinced my company to do the same on the past. But I have a very hard time to convince people to keep using it every time they argue with me "why such obvious things [like basic focus/blur/keyboard control or at least basic events] is not there by default?""

I think there should be a common foundation of consistent functionality and events built in and reliable found on every component across the board. 

You should be able to command "focus" to any visual component and that component should activate its "focused state" no matter how it renders itself on screen. 

You should be able to intercept any normal event like keys and mouse related on any iterable component no matter how it is rendered, even if one event or another in particular was not natively present by the way some specific comp is build, and in this case it should be fired by emulation just to keep the framework concise.

Its a small concern, but it becomes very hard when you need to build things on interfaces that demand a lot of custom behaviour.

Thanks!

**Comment by hawkeye64** [comment, marked as answer]
This is a bit nonsensical. Quasar does not prevent any of this, nor does Vue. In the end, a compiled component is just an element or group of elements.

**Reply by labs20** [reply]
Hi! Thanks for your answer.

Focus the radio and then press ENTER or SPACE, you will see that is not logs to the console, while other keys do. Event on those keys are not being propagated.

```

<template lang="pug">
q-option-group(type="radio" v-model="radio" :options="opts" @keyup="keyUp")
</template>

<script>
import { defineComponent, ref, reactive } from 'vue';
export default defineComponent({
    name: "Test",
    setup(props, context){
        const radio = ref(1)
            , opts = reactive([
                {value:1, label: 'One'},
                {value:2, label: 'Two'},
                {value:3, label: 'Three'}
            ])
        ;
        return {
            radio, opts
        }
    },
    methods: {
        keyUp(ev){
            console.log(ev.key);
        }
    }
});
</script>

```

**Reply by pdanpdan** [reply]
I would suggest you should use `.capture` for such events - there are:
- cases where we don't want to allow the event to propagate up to prevent the havoc that would happen (these are like this to stay)
- cases where we forgot to propagate - if you find things like this please tell us to fix them

If you want you can help us by checking where `stop` or `stopAndPrevent` is used in the Quasar codebase in the context of keyboard events and make a list so we can check.

**Reply by labs20** [reply]
`I would suggest you should use .capture for such events`
Yeap! It works! Thank you very much.

...but, I'll keep my suggestion about that common ground set of features, and if I may suggest, this `.capture` hint would fit fine at the docs page.

Thank you again for your help.
